### DISCUSSION TITLE
*SOLVED* How do you get regression coefficients and their significance from PolynomialTrendForecaster?

### DISCUSSION THREAD

**Discussion by ericjb** [alan-turing-institute/sktime#5743] [Q&A] [answered]
After fitting a PolynomialTrendForecaster, how can I see the fitted coefficients, their significance, and other info? Happy to get a pointer to the relevant places in the documentation or a tutorial. Thanks

**Comment by fkiraly** [comment]
You can access the regressor in the `regressor_` attribute - this was not documented unforuntately, but I added it in this PR https://github.com/sktime/sktime/pull/5747

In general, you can get fitted parameters (that the estimator exposes) via `get_fitted_params`, this works for all `scikit-base` object, all `sktime` objects are `scikit-base` objects.

For significances, you would have to use the summary of the fitted estimator, which you can get via `regressor_` attribute. As far as I know, `sklearn`'s linear regression does *not* expose a classical statistics summary, unlike `statsmodels` `OLS` for instance.

A note of caution, as you are dealing with time series which are temporally correlated, the significances will be misleading, as central assumptions - such as i.i.d.-ness of (conditional) samples, or error terms - will not be satisfied.

**Reply by ericjb** [reply]
I understand that sklearn's linear regression does not expose a summary() method. Can you explain how to use get_fitted_params (or any other technique) to locate the fitted coefficients of the regression? Seems strange to me that such an important and basic method as linear regression would not provide access - even if through some effort - to the coefficients and their significance.  I am trying to convert my workflow for time series related projects to sktime but am frustrated by how difficult this piece is (especially by comparison to how easy other aspects are - thank you for your efforts FK)

**Reply by fkiraly** [reply]
the coefficients are exposed by sklearn, the attribute is `coef_`.

Unfortunately sklearn does not have `get_fitted_params`, so you have to check the docs.

**Comment by ericjb** [comment, marked as answer]
I found a solution/workaround using ARDL which has a summary() method.

'
## Do a quadratic linear regression, i.e. (1, t, t**2)
## Show the regression coefficients with statistical significance
from sktime.forecasting.ardl import ARDL
from sktime.split import temporal_train_test_split
t = np.arange(1, len(y)+1)
columns = [pow(t,i) for i in range(3)]
X = pd.DataFrame(np.vstack(columns).T, index=y.index)
X.columns = ["const", "t", "t^2"]
ardl_quad = ARDL(lags=0, order=0, trend='n', seasonal=False, auto_ardl=False)
y_train, y_valid = temporal_train_test_split(y)
ardl_quad.fit(y_train, X=X.iloc[:len(y_train),:])
print(ardl_quad.summary())
'

**Reply by ericjb** [reply]
In my submitted code the exponentiation operation * * (with no space) was interpreted as start/end of boldface. :-(

**Reply by fkiraly** [reply]
if you surround it by a tick, it will render as code

**Reply by ericjb** [reply]
I replaced t**i by pow(t,i) to avoid the markdown conversions to bold that corrupted the meaning.
